csm_web/scheduler/admin.py

```diff
@@ -49,273 +96,311 @@ def has_delete_permission(self, request, obj=None):
         return request.user.is_superuser
 
 
+@admin.register(Student)
+class StudentAdmin(CoordAdmin):
```

Comment:
Because you have `get_email`, `get_course`, `get_mentor`, and `section` in `list_display`, the admin will have to make a SQL query for each of the related objects for each Student, which is **terrible** for performance, even on a small dataset you can overwhelm the connection pool like this.

To remedy this, preload the relations for each object in the query set, which drastically reduces the number of queries made:
```python
def get_queryset(self, request):
    return super().get_queryset(request).select_related('user', 'section__course', 'section__mentor')
```